# Individual Plan  

github link: https://github.com/ansonchan031021-source/DSCI_100_individual_plan.git

In [17]:
library(tidyverse)
url_1 = 'https://drive.google.com/uc?export=download&id=1hSH3Uxef4JcuooVcPXiY0wfRlW0Qahbs'
url_2 = 'https://drive.google.com/uc?export=download&id=1SR_EZnftKBWy8MH-1t2wBtRRrr4DJjmn'

players <- read_csv(url_1)
dim(players); head(players)

sessions <- read_csv(url_2)
dim(sessions); head(sessions)

Rows: 196 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): experience, hashedEmail, name, gender
dbl (2): played_hours, Age
lgl (1): subscribe

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


[1] 196   7

experience subscribe
1 Pro         TRUE    
2 Veteran     TRUE    
3 Veteran    FALSE    
4 Amateur     TRUE    
5 Regular     TRUE    
6 Amateur     TRUE    
  hashedEmail                                                      played_hours
1 f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d 30.3        
2 f3c813577c458ba0dfef80996f8f32c93b6e8af1fa939732842f2312358a88e9  3.8        
3 b674dd7ee0d24096d1c019615ce4d12b20fcbff12d79d3c5a9d2118eb7ccbb28  0.0        
4 23fe711e0e3b77f1da7aa221ab1192afe21648d47d2b4fa7a5a659ff443a0eb5  0.7        
5 7dc01f10bf20671ecfccdac23812b1b415acd42c2147cb0af4d48fcce2420f3e  0.1        
6 f58aad5996a435f16b0284a3b267f973f9af99e7a89bee0430055a44fa92f977  0.0        
  name      gender Age
1 Morgan    Male    9 
2 Christian Male   17 
3 Blake     Male   17 
4 Flora     Female 21 
5 Kylie     Male   21 
6 Adrian    Female 17

Rows: 1535 Columns: 5
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (3): hashedEmail, start_time, end_time
dbl (2): original_start_time, original_end_time

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


[1] 1535    5

hashedEmail                                                     
1 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
2 36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
3 f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
4 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
5 36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
6 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
  start_time       end_time         original_start_time original_end_time
1 30/06/2024 18:12 30/06/2024 18:24 1.71977e+12         1.71977e+12      
2 17/06/2024 23:33 17/06/2024 23:46 1.71867e+12         1.71867e+12      
3 25/07/2024 17:34 25/07/2024 17:57 1.72193e+12         1.72193e+12      
4 25/07/2024 03:22 25/07/2024 03:58 1.72188e+12         1.72188e+12      
5 25/05/2024 16:01 25/05/2024 16:12 1.71665e+12         1.71665e+12      
6 23/06/2024 15:08 23/06/2024 17:10 1.71916e+12         1.71916e+12

In [18]:
distinct(players, experience)
distinct(players, subscribe)
distinct(players, gender)

players|>
    summarize(max_played = max(played_hours, na.rm = TRUE),
             min_played = min(played_hours, na.rm = TRUE),
             max_age = max(Age, na.rm = TRUE),
             min_age = min(Age, na.rm = TRUE))

experience
1 Pro       
2 Veteran   
3 Amateur   
4 Regular   
5 Beginner

subscribe
1  TRUE    
2 FALSE

gender           
1 Male             
2 Female           
3 Non-binary       
4 Prefer not to say
5 Agender          
6 Two-Spirited     
7 Other

max_played min_played max_age min_age
1 223.1      0          58      9

# Part 1: Data Description  

#### There are 196 observations and 9 variables in the player table, the variables are:  

+ `experience`: charactor, it shows that the variable experience has five unique categories — `Pro`, `Veteran`, `Amateur`, `Regular`, and `Beginner` — representing different levels of player experience.
+ `subscribe`: logical, it shows whether the player subscribed to the game or not in `TRUE` or `FALSE`
+ `hashedEmail`: character, it stores anonymized user email identifiers in a hashed format to protect personal information.
+ `played_hours	`: double, it records the game playtime in hours, with a range from 0 to 223.1.
+ `name`: character, it shows different players' names.
+ `gender`: charactor, it shows the variable genders have 7 unique categories - `Male`, `Female`, `Non-binary`, `Prefer not to say`, `Agender`, `Two-Spirited` and `other`. 
+ `Age`: double, it records the players' age, with a range of 9 to 58.

#### There are 1535 observations and 5 variables in the sessions table, the variables are:  

+ `hashedEmail`: character, it stores anonymized user email identifiers in a hashed format to protect personal information.
+ `start_time`: character, it shows the game start time, each entry follows the date-time format DD/MM/YYYY (24-hour clock).
+ `end_time`: character, it shows the game end time, each entry follows the date-time format DD/MM/YYYY (24-hour clock).
+ `original_start_time`: double, it is the original data and is not human-readable, so it cannot be used directly.
+ `original_end_time`: double, it is also the original data and is not human-readable, so it cannot be used directly.

### Potential issues

In [19]:
players |> group_by(Age)|> summarize(count = n())

Age count
1   9   1   
2  10   1   
3  11   1   
4  12   1   
5  14   2   
6  15   2   
7  16   3   
8  17  73   
9  18   7   
10 19   7   
11 20  14   
12 21  18   
13 22  15   
14 23  13   
15 24  10   
16 25   3   
17 26   4   
18 27   3   
19 28   2   
20 32   2   
21 33   1   
22 37   1   
23 38   1   
24 42   1   
25 44   1   
26 45   1   
27 46   1   
28 47   1   
29 49   1   
30 50   1   
31 57   1   
32 58   1   
33 NA   2

1. The data is mainly concentrated in the age range of 16 to 27. Therefore, the accuracy rate of the data outside this range may decline. In the prediction process, if the accuracy rate is too low, we can choose to keep this part of the data and remove the data from other age groups, and then see if we can improve the accuracy rate.
2. Data such as hashedEmail, original_start_time and original_end_time cannot be read and thus cannot be used.

# Part 2: Question

I will choose  **question 1** to answer:  What player characteristics and behaviours are most predictive of subscribing to a game-related newsletter, and how do these features differ between various player types?  
**specific question**: Can player `experience`, `played_hour`s, and `age` predict whether a player subscribes to a game-related newsletter? 